bebraven/platform
# Deprecate Registration Callback On User Model

### Description
This PR moves the account registration logic away from the user model.

PS: I intend to have a follow up PR that'll clean up any dead code.

### Motivation and Context
- Improve code quality. 
- Separate portal account creation steps into services so that they can be later run independently. 
- Reduce synchronous process during sign up

https://app.asana.com/0/1179330853755533/1189915592902033
Part of: https://app.asana.com/0/1179330853755533/1189555546176041

### Types of changes




### How to test
- Try sign up with an ID from Salesforce. You should get an email response.
- Try to sign up when you already have an account.
- Try to run sync to LMS using ID from Salesforce. You should get an email response


### Checklist:
